core/indexer/processTransactions.go

```diff
@@ -118,8 +130,20 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 	return append(convertMapTxsToSlice(transactions), rewardsTxs...)
 }
 
-func (tdp *txDatabaseProcessor) addScResultInfoInTx(scr *smartContractResult.SmartContractResult, tx *Transaction) *Transaction {
-	dbScResult := tdp.commonProcessor.convertScResultInDatabaseScr(scr)
+func findAllChildScrResults(hash string, scrs map[string]*smartContractResult.SmartContractResult) map[string]*smartContractResult.SmartContractResult {
```

Comment:
add unit tests for intrashard and cross shard relayed smart contracts. verify how those are indexed.

Reply:
done. added a unit tests and did a lot of manual test to check if transaction are indexed correctly

Reply:
maybe rename it as this function also removes elements from the map received as parameter